src/ai/backend/manager/api/exceptions.py

```diff
@@ -412,3 +412,8 @@ class KernelRestartFailed(BackendAgentError, web.HTTPInternalServerError):
 class KernelExecutionFailed(BackendAgentError, web.HTTPInternalServerError):
     error_type  = 'https://api.backend.ai/probs/kernel-execution-failed'
     error_title = 'Executing user code in the kernel has failed.'
+
+
+class UnknownImageReferenceError(BackendError, web.HTTPBadRequest):
```

Comment:
Please inherit `ObjectNotFound` instead of generic `BackendError` so that the status code becomes 404.

Reply:
Fixed it.